env0/resource_environment.go

```diff
@@ -653,3 +682,28 @@ func resourceEnvironmentImport(ctx context.Context, d *schema.ResourceData, meta
 		return []*schema.ResourceData{d}, nil
 	}
 }
+
+func waitForDeployment(d *schema.ResourceData, apiClient client.ApiClientInterface) error {
+	for {
+		env, err := apiClient.Environment(d.Id())
+		if err != nil {
+			return err
+		}
+		switch env.LatestDeploymentLog.Status {
+		case "IN_PROGRESS",
+			"QUEUED",
+			"WAITING_FOR_USER":
+			// Deployment still in progress - wait 1 second then check again
```

Comment:
I think the comments in this function aren't really needed, it's clear enough as it is